bigframes/core/compile/compiled.py

```diff
@@ -164,112 +331,108 @@ def mem_expr_from_pandas(
                 keys_memtable[f"col_{column_index}"].name(column)
                 for column_index, column in enumerate(column_names)
             ],
-            ordering=ExpressionOrdering(
-                ordering_value_columns=tuple(
-                    [OrderingColumnReference(ORDER_ID_COLUMN)]
-                ),
-                total_ordering_columns=frozenset([ORDER_ID_COLUMN]),
-            ),
-            hidden_ordering_columns=(keys_memtable[ORDER_ID_COLUMN],),
         )
 
-    @property
-    def columns(self) -> typing.Tuple[ibis_types.Value, ...]:
-        return self._columns
+    def builder(self):
+        """Creates a mutable builder for expressions."""
+        # Since ArrayValue is intended to be immutable (immutability offers
+        # potential opportunities for caching, though we might need to introduce
+        # more node types for that to be useful), we create a builder class.
+        return UnorderedIR.Builder(
+            self._table,
+            columns=self._columns,
+            predicates=self._predicates,
+        )
 
-    @property
-    def column_ids(self) -> typing.Sequence[str]:
-        return tuple(self._column_names.keys())
-
-    @property
-    def _hidden_column_ids(self) -> typing.Sequence[str]:
-        return tuple(self._hidden_ordering_column_names.keys())
-
-    @property
-    def _reduced_predicate(self) -> typing.Optional[ibis_types.BooleanValue]:
-        """Returns the frame's predicates as an equivalent boolean value, useful where a single predicate value is preferred."""
-        return (
-            _reduce_predicate_list(self._predicates).name(PREDICATE_COLUMN)
-            if self._predicates
-            else None
+    def to_sql(
+        self,
+        offset_column: typing.Optional[str] = None,
+        col_id_overrides: typing.Mapping[str, str] = {},
+        sorted: bool = False,
+    ) -> str:
+        if offset_column or sorted:
+            raise ValueError("Cannot produce sorted sql in unordered mode")
+        sql = ibis_bigquery.Backend().compile(
+            self._to_ibis_expr(
+                col_id_overrides=col_id_overrides,
+            )
         )
+        return typing.cast(str, sql)
 
-    @property
-    def _ibis_order(self) -> Sequence[ibis_types.Value]:
-        """Returns a sequence of ibis values which can be directly used to order a table expression. Has direction modifiers applied."""
-        return _convert_ordering_to_table_values(
-            {**self._column_names, **self._hidden_ordering_column_names},
-            self._ordering.all_ordering_columns,
-        )
+    def _to_ibis_expr(
+        self,
+        *,
+        expose_hidden_cols: bool = False,
```

Comment:
Might be able to remove this parameter.

Reply:
This parameter is a bit tricky, as its mostly there to prevent ibis from doing to many re-projections. I don't want to risk removing this in this cl